json_schemas/core.schema.json

```diff
@@ -0,0 +1,185 @@
+    "definitions": {
+        "did.schema" : {
+            "type" : "string",
+            "pattern" : "^did:[a-z0-9]+:[a-zA-Z0-9.-_:]+"
+        },
+
+        "type.schema" : {
+            "type" : [
+                "string",
+                "array"
+            ],
+            "uniqueItems": true,
+            "items" : {
+                "type" : "string"
+            }
+        },
+
+        "JWK.schema" : {
+            "type" : "object",
+            "properties": {
+                "kty": {
+                    "type": "string"
+                },
+                "use": {
+                    "type": "string"
+                },
+                "key_ops": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "alg": {
+                    "type": "string"
+                },
+                "kid": {
+                    "type": "string"
+                },
+                "x5u": {
+                    "type": "string"
+                },
+                "x5c": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "x5t": {
+                    "type": "string"
+                },
+                "x5t#S256": {
+                    "type": "string"
+                }
+            },
+            "required": ["kty"],
+            "additionalProperties": true
+        },
+
+        "verification_method.schema" : {
+            "type" : "array",
+            "uniqueItems": true,
+            "items" : {
+                "type" : "object",
+                "properties": {
+                    "publicKeyJwk" : {
+                        "$ref" : "#/definitions/JWK.schema"
+                    },
+                    "publicKeyBase58" : {
+                        "type" : "string"
+                    },
+                    "controller" : {
+                        "$ref" : "#/definitions/did.schema"
+                    },
+                    "id" : {
+                        "type": "string",
+                        "format" : "uri-reference"
+                    },
+                    "type" : {
+                        "$ref" : "#/definitions/type.schema"
+                    }
+                },
+                "required" : ["id", "type", "controller"],
+                "oneOf": [
+                    {"required" : ["publicKeyJwk"]},
+                    {"required" : ["publicKeyBase58"]}
+                ],
+                "additionalProperties" : true
+            }
+        },
+
+        "service.schema" : {
+            "type" : "array",
+            "uniqueItems": true,
+            "items" : {
+                "type" : "object",
+                "properties" : {
+                    "id" : {
```

Comment:
@csuwildcat if you want cannonical or equivalentId to be included in this, you may need to open a new PR in the future.

Reply:
canonicalID and equivalentId are metadata properties now right? So wouldn't be covered by this

Reply:
@rhiaro @iherman both are present in the DID Document as "INFRA MAP keys...." See the existing tests for them:

https://github.com/w3c/did-test-suite/blob/master/packages/did-core-test-server/suites/did-spec/default.json#L23